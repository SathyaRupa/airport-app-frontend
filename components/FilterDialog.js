import React, {useState} from 'react';
import {Portal, Dialog, Text, Button} from 'react-native-paper';
import {View, TextInput, StyleSheet} from 'react-native';

export default function FilterDialog({
  visible,
  dismiss,
  onClick,
  setFloor,
  floor,
}) {
  const [floorError, setFloorError] = useState('');
  function handleFloorChange(newText) {
    if (!/^\d*$/.test(newText)) {
      setFloorError('Floor number must be a number.');
    } else if (parseInt(newText) > 10) {
      setFloorError('Floor number should not exceed 10.');
    } else {
      setFloorError('');
    }
    setFloor(newText);
  }
  return (
    <Portal>
      <Dialog visible={visible} onDismiss={dismiss} style={styles.dialog}>
        <Dialog.Content>
          <Text>Floor number :</Text>
          <TextInput
            style={styles.input}
            value={floor}
            onChangeText={handleFloorChange}
            testID="floor-input-column"
          />
          {floorError ? (
            <Text style={styles.errorText}>{floorError}</Text>
          ) : (
            <Text> </Text>
          )}
        </Dialog.Content>
        <Dialog.Actions>
          <Button
            mode="text"
            disabled={!!floorError}
            onPress={onClick}
            style={styles.dialogButton}>
            Done
          </Button>
        </Dialog.Actions>
      </Dialog>
    </Portal>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  dialog: {
    backgroundColor: 'white',
    padding: 20,
    height: 190,
    width: 290,
    alignSelf: 'center',
  },
  input: {
    borderBottomColor: 'black',
    borderBottomWidth: 1,
    marginBottom: 5,
    paddingHorizontal: 8,
    paddingVertical: 4,
  },
  errorText: {
    color: 'red',
    marginTop: 5,
    fontSize: 11,
  },
  dialogButton: {
    marginBottom: 5,
  },
});
