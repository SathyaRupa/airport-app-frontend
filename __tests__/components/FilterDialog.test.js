import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import {Provider as PaperProvider} from 'react-native-paper';
import FilterDialog from '../../components/FilterDialog';

describe('FilterDialog', () => {
  let props;

  beforeEach(() => {
    props = {
      visible: true,
      dismiss: jest.fn(),
      onClick: jest.fn(),
      floor: '',
      floorError: '',
      setFloor: jest.fn(),
      setFloorError: jest.fn(),
    };
  });

  const renderComponent = (overrideProps = {}) => {
    const combinedProps = {...props, ...overrideProps};
    return render(
      <PaperProvider>
        <FilterDialog {...combinedProps} />
      </PaperProvider>,
    );
  };
  it('snapshot testing', () => {
    const filterDialog = renderComponent();
    expect(filterDialog).toMatchSnapshot();
  });
  it('renders correctly', () => {
    const {getByText, getByTestId} = renderComponent();
    expect(getByText('Floor number :')).toBeTruthy();
    expect(getByTestId('floor-input-column')).toBeTruthy();
  });

  it('displays error message for invalid floor number', () => {
    const {getByTestId, getByText} = renderComponent();
    const input = getByTestId('floor-input-column');

    fireEvent.changeText(input, 'abc');
    expect(getByText('Floor number must be a number.')).toBeTruthy();

    fireEvent.changeText(input, '11');
    expect(getByText('Floor number should not exceed 10.')).toBeTruthy();
  });

  it('disables the Done button when there is an error', () => {
    const {getByText, getByTestId} = renderComponent();
    const input = getByTestId('floor-input-column');

    fireEvent.changeText(input, 'abc');
    const doneButton = getByText('Done');

    fireEvent.press(doneButton);
    expect(props.onClick).not.toHaveBeenCalled();
  });

  it('enables the Done button when there is no error', () => {
    const {getByText, getByTestId} = renderComponent();
    const input = getByTestId('floor-input-column');

    fireEvent.changeText(input, '6');
    const doneButton = getByText('Done');

    fireEvent.press(doneButton);
    expect(props.onClick).toHaveBeenCalled();
  });

  it('calls onClick when Done button is pressed', () => {
    const {getByText} = renderComponent({floorError: ''});
    const doneButton = getByText('Done');

    fireEvent.press(doneButton);
    expect(props.onClick).toHaveBeenCalled();
  });
});
