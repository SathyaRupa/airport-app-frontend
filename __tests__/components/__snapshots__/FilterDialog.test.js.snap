// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterDialog snapshot testing 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  />
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      accessibilityLiveRegion="polite"
      accessibilityViewIsModal={true}
      collapsable={false}
      onAccessibilityEscape={[Function]}
      pointerEvents="auto"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      testID="modal"
    >
      <View
        accessibilityLabel="Close modal"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        importantForAccessibility="no"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "rgba(50, 47, 55, 0.4)",
            "flex": 1,
            "opacity": 1,
          }
        }
        testID="modal-backdrop"
      />
      <View
        pointerEvents="box-none"
        style={
          [
            {
              "bottom": 0,
              "justifyContent": "center",
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            {
              "marginBottom": 0,
              "marginTop": 0,
            },
            undefined,
          ]
        }
        testID="modal-wrapper"
      >
        <View
          collapsable={false}
          style={
            {
              "alignSelf": "center",
              "backgroundColor": "white",
              "borderRadius": 28,
              "height": 190,
              "marginHorizontal": 26,
              "marginVertical": 0,
              "opacity": 1,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.15,
              "shadowRadius": 3,
              "width": 290,
            }
          }
          testID="modal-surface-outer-layer"
        >
          <View
            collapsable={false}
            style={
              {
                "backgroundColor": "white",
                "borderRadius": 28,
                "elevation": 24,
                "flex": 1,
                "justifyContent": "flex-start",
                "padding": 20,
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 1,
                  "width": 0,
                },
                "shadowOpacity": 0.3,
                "shadowRadius": 1,
              }
            }
            testID="modal-surface"
          >
            <View
              style={
                [
                  {
                    "paddingBottom": 24,
                    "paddingHorizontal": 24,
                  },
                  [
                    {
                      "marginTop": 24,
                    },
                    undefined,
                  ],
                ]
              }
            >
              <Text
                style={
                  [
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontFamily": "System",
                      "fontWeight": "400",
                      "letterSpacing": 0,
                    },
                    {
                      "writingDirection": "ltr",
                    },
                    undefined,
                  ]
                }
              >
                Floor number :
              </Text>
              <TextInput
                onChangeText={[Function]}
                style={
                  {
                    "borderBottomColor": "black",
                    "borderBottomWidth": 1,
                    "marginBottom": 5,
                    "paddingHorizontal": 8,
                    "paddingVertical": 4,
                  }
                }
                testID="floor-input-column"
                value=""
              />
              <Text
                style={
                  [
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontFamily": "System",
                      "fontWeight": "400",
                      "letterSpacing": 0,
                    },
                    {
                      "writingDirection": "ltr",
                    },
                    undefined,
                  ]
                }
              >
                 
              </Text>
            </View>
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "flexGrow": 1,
                    "justifyContent": "flex-end",
                    "paddingBottom": 24,
                    "paddingHorizontal": 24,
                  },
                  undefined,
                ]
              }
            >
              <View
                collapsable={false}
                style={
                  {
                    "backgroundColor": "transparent",
                    "borderRadius": 20,
                    "marginBottom": 5,
                    "marginRight": 0,
                    "shadowColor": "#000",
                    "shadowOffset": {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0,
                    "shadowRadius": 0,
                  }
                }
                testID="button-container-outer-layer"
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "backgroundColor": "transparent",
                      "borderColor": "transparent",
                      "borderRadius": 20,
                      "borderStyle": "solid",
                      "borderWidth": 0,
                      "flex": undefined,
                      "minWidth": "auto",
                      "shadowColor": "#000",
                      "shadowOffset": {
                        "height": 0,
                        "width": 0,
                      },
                      "shadowOpacity": 0,
                      "shadowRadius": 0,
                    }
                  }
                  testID="button-container"
                >
                  <View
                    accessibilityRole="button"
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": false,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      [
                        {
                          "overflow": "hidden",
                        },
                        {
                          "borderRadius": 20,
                        },
                      ]
                    }
                    testID="button"
                  >
                    <View
                      style={
                        [
                          {
                            "alignItems": "center",
                            "flexDirection": "row",
                            "justifyContent": "center",
                          },
                          undefined,
                        ]
                      }
                    >
                      <Text
                        numberOfLines={1}
                        selectable={false}
                        style={
                          [
                            {
                              "textAlign": "left",
                            },
                            {
                              "color": "rgba(28, 27, 31, 1)",
                              "writingDirection": "ltr",
                            },
                            [
                              {
                                "fontFamily": "System",
                                "fontSize": 14,
                                "fontWeight": "500",
                                "letterSpacing": 0.1,
                                "lineHeight": 20,
                              },
                              [
                                {
                                  "marginHorizontal": 16,
                                  "marginVertical": 9,
                                  "textAlign": "center",
                                },
                                false,
                                {
                                  "marginHorizontal": 12,
                                },
                                {
                                  "marginHorizontal": 8,
                                },
                                false,
                                {
                                  "color": "rgba(103, 80, 164, 1)",
                                  "fontFamily": "System",
                                  "fontSize": 14,
                                  "fontWeight": "500",
                                  "letterSpacing": 0.1,
                                  "lineHeight": 20,
                                },
                                undefined,
                              ],
                            ],
                          ]
                        }
                        testID="button-text"
                      >
                        Done
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
